pkg/cli/image/manifest/manifest.go

```diff
@@ -109,9 +113,18 @@ func (o *SecurityOptions) NewContext() (*registryclient.Context, error) {
 			return nil, fmt.Errorf("unable to load --registry-config: %v", err)
 		}
 	}
-	context := registryclient.NewContext(rt, insecureRT).WithCredentials(creds)
-	context.DisableDigestVerification = o.SkipVerification
-	return context, nil
+	pushCreds := dockercredentials.NewLocal()
+	if len(o.PushConfig) > 0 {
+		pushCreds, err = dockercredentials.NewFromFile(o.PushConfig)
```

Comment:
haven't looked in great detail but it seems to return a custom implementation of a store that relies on a `credentialprovider.DockerKeyring` with the following interface:

```
// DockerKeyring tracks a set of docker registry credentials, maintaining a
// reverse index across the registry endpoints. A registry endpoint is made
// up of a host (e.g. registry.example.com), but it may also contain a path
// (e.g. registry.example.com/foo) This index is important for two reasons:
// - registry endpoints may overlap, and when this happens we must find the
//   most specific match for a given image
// - iterating a map does not yield predictable results
type DockerKeyring interface {
	Lookup(image string) ([]AuthConfig, bool)
}
```

it looks like we could actually find the most specific match for a given image, wdyt?

Reply:
i will try : ) will report back